src/Symfony/Component/DependencyInjection/Compiler/AnalyzeServiceReferencesPass.php

```diff
@@ -136,15 +136,27 @@ protected function processValue($value, $isRoot = false)
         }
         $this->lazy = false;
 
+        // Any calls before a "wither" are part of the constructor-instantiation graph
```

Comment:
Aren't any call to a non-wither performed before a wither call also part of it ?

Reply:
Please add some tests covering mixing wither calls with non-wither calls (in different orders) to make sure this works right.

Reply:
I would argue that non-wither calls should happen after wither calls, on the off chance that some calls register themselves with other services and the object hashes differ and causes unintended problems.

Reply:
Well, I would say that mixing things are an edge case indeed. But the current configuration syntax supports it, and so we should deal with it properly.

The code currently already respects the order of method calls, be them withers or no, when instantiating the service. Only this place is missing that.

Reply:
> Aren't any call to a non-wither performed before a wither call also part of it ?

sure, that'swhat I mean by "any calls" - regular setters or withers

> non-wither calls should happen after wither calls

why not, but not something I would make a "must"